power_grid_model_c/power_grid_model/include/power_grid_model/component/branch.hpp

```diff
@@ -37,11 +37,21 @@ class Branch : public Base {
     }
 
     // getter
-    ID from_node() const { return from_node_; }
-    ID to_node() const { return to_node_; }
-    bool from_status() const { return from_status_; }
-    bool to_status() const { return to_status_; }
-    bool branch_status() const { return from_status_ && to_status_; }
+    constexpr ID from_node() const { return from_node_; }
+    constexpr ID to_node() const { return to_node_; }
+    constexpr bool from_status() const { return from_status_; }
+    constexpr bool to_status() const { return to_status_; }
+    constexpr bool branch_status() const { return from_status_ && to_status_; }
+    constexpr bool status(BranchSide side) const {
+        switch (side) {
+        case BranchSide::from:
+            return from_status_;
+        case BranchSide::to:
+            return to_status_;
+        default:
+            throw MissingCaseForEnumError{"connected_at_side<BranchSide>", side};
+        }
+    }
```

Comment:
this type of code is actually needed fairly often in the tap regulator. to prevent the need of having to use a `switch`/`case` statement with separate calls to `status_from()` and `status_to()` every single time, I thought it would be a good idea to add this.

what do you need with `node` call? similar functionality to obtain the `node` from `BranchSide side`?

Reply:
Indeed, is there a logic which will get replaced?

Reply:
>  to prevent the need of having to use a switch/case statement with separate calls to status_from() and status_to() every single time, I thought it would be a good idea to add this.

Wouldn't this be redundant?

> what do you need with node call? similar functionality to obtain the node from BranchSide side?

Now that you might switch calls to `status_from()` and `status_to()`, with the same logic, why wouldn't there be switch calls to `node_from()` and `node_to()`?

Reply:
>> to prevent the need of having to use a switch/case statement with separate calls to status_from() and status_to() every single time, I thought it would be a good idea to add this.
> Wouldn't this be redundant?

maybe i don't understand what you meant in the first place. can you elaborate a bit maybe?



>> what do you need with node call? similar functionality to obtain the node from BranchSide side?
> Now that you might switch calls to status_from() and status_to(), with the same logic, why wouldn't there be switch calls to node_from() and node_to()?

i had to look it up whether this really was useful but turns out that it is. will do

Reply:
>maybe i don't understand what you meant in the first place. can you elaborate a bit maybe?

It's very minor, let's ignore it for now